dango/dex/src/state.rs

```diff
@@ -1,18 +1,16 @@
 use {
-    dango_types::dex::{Direction, OrderId, PairParams},
-    grug::{Addr, Counter, Counters, Denom, IndexedMap, Map, Udec128, Uint128, UniqueIndex},
+    dango_types::{
+        dex::{Direction, OrderId, PairParams},
+        TransientMap,
+    },
+    grug::{Addr, Counter, Denom, IndexedMap, Map, Udec128, Uint128, UniqueIndex},
 };
 
+pub(crate) static INCOMING_ORDERS: TransientMap<OrderId, (OrderKey, Order)> = TransientMap::new();
```

Comment:
Noticed an issue: this static variable is shared by all threads. Meaning if one simulates a tx using the query thread, a new order would be inserted into the map, impacting the block execution thread. A solution may be to use the `thread_local!` macro.